### ISSUE TITLE
The shape of label tensor for CosineEmbeddingLoss should be specified in the document

### ISSUE THREAD

**Issue by yistLin** [pytorch/pytorch#52732]
## 📚 Documentation

I'm now using torch 1.7.1. And in the [document for CosineEmbeddingLoss](https://pytorch.org/docs/stable/generated/torch.nn.CosineEmbeddingLoss.html#torch.nn.CosineEmbeddingLoss), it should be mentioned that the label tensor **y** has to be **a 1D mini-batch Tensor**. Otherwise, even though the label tensor contains  only -1 and 1, the loss value would be incorrect if the tensor is a 2D mini-batch Tensor.

For example:
```python
import torch
import torch.nn as nn

torch.manual_seed(0)

a = torch.randn(16, 5)
b = torch.randn(16, 5)
labels = torch.empty(16).bernoulli_().mul_(2).sub_(1)

loss_fn = nn.CosineEmbeddingLoss()

print(loss_fn(a, b, labels))  # 0.4538
print(loss_fn(a, b, labels.unsqueeze(1)))  # 0.4810
```

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @brianjo @mruberry @albanD

**Comment by mruberry** [comment]
@jbschlosser to consider the documentation change

**Comment by jbschlosser** [comment]
The documentation should certainly be updated to include the expected input and target shapes, as is the case for most other loss functions (e.g. [TripletMarginWithDistanceLoss](https://pytorch.org/docs/stable/generated/torch.nn.TripletMarginWithDistanceLoss.html)).

(Side note: [MultiMarginLoss](https://pytorch.org/docs/stable/generated/torch.nn.MultiMarginLoss.html) is another that doesn't and should also be updated.)

That said, I think this should also qualify as a bug, as the shape difference should not change the result. The usage of `where()` in the implementation of `CosineEmbeddingLoss` is the issue- its broadcasting behavior leads to the incorrect computation.

**Comment by jbschlosser** [comment]
Opened a separate issue for the bug: #53030

**Comment by jbschlosser** [comment]
Removing high-pri for the doc update portion.
